massa-models/src/denunciation.rs

```diff
@@ -714,6 +755,93 @@ impl Deserializer<Denunciation> for DenunciationDeserializer {
 
 // End Ser / Der
 
+// Denunciation Id
+
+/// Endorsement ID size in bytes
+pub const DENUNCIATION_ID_SIZE_BYTES: usize = massa_hash::HASH_SIZE_BYTES;
+
+/// endorsement id
+#[derive(Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct DenunciationId(Hash);
+
+const DENUNCIATION_ID_PREFIX: char = 'D';
+const DENUNCIATION_ID_VERSION: u64 = 0;
+
+impl PreHashed for DenunciationId {}
+
+impl Id for DenunciationId {
```

Comment:
I think that the whole concept of DenunciationID is not necessary. All denunciations targeting a given slot (or pair slot/index) are identical

Reply:
It is used in DenunciationPool in order to have a common hashmap key as it covers both use case (e.g. for block header denunciation & endorsement denunciation).

Reply:
Can't we just use the following locally in the pool ?

```rust
#[derive(Debug, Copy, Hash, PartialOrd, Ord, PartialEq, Eq)]
pub struct DenunciationKey {
    Header {
       slot: Slot
    },
    Endorsement {
        slot: Slot,
        index: Index
    }
}
```

that way we don't need to hash, nor to implement serialization, deserialization etc...

Reply:
I've added a TODO to change this

Reply:
Note: this key will also be used in ExecutedDenunciations